### ISSUE TITLE
Tween can't properly animate Matrix32

### ISSUE THREAD

**Issue by bojidar-bg** [godotengine/godot#2627]
As said, `Tween` can't really animate Matrix32 transformations, but instead converts them into Matrix3.

I think the problem comes from [tween.cpp:407-409](https://github.com/okamstudio/godot/blob/master/scene/animation/tween.cpp#L408), where the `Matrix3` class is used instead of the proper `Matrix32`

**Comment by romulox-x** [comment]
That does seems suspicious.  Though there could be some automatic casting back to Matrix32 that its relying on.

I'm not very faimiliar with the tween class, but...is that just interpolating between matrix elements?  That seems weird.  Won't that generate wacky transformations in between?  Maybe you're not meant to use this for transformations directly, and only when you actually want to interpolate between the elements for some reason?

**Comment by bojidar-bg** [comment]
I can confirm that there is *no* auto-casing into Matrix32.
````python
func _ready():
  get_node("Tween").interpolate_method( # newlines added for readability
    self, 
    "print_type",
    Matrix32(90, Vector2(1,0)),
    Matrix32(0, Vector2(0,0)),
    1,
    Tween.TRANS_LINEAR, Tween.EASE_IN)
  get_node("Tween").start()
func print_type(v):
  prints(typeof(v), TYPE_MATRIX32, TYPE_MATRIX3)
````
Prints:
````
8 8 12
12 8 12
12 8 12
...
````

@romulox-x ~ There is a `Matrix32::interpolate_with` that seems to be a better idea in this case...

**Comment by reduz** [comment]
the tween class is kind of a weird contribution, it works really odd in some cases and probably should be given some rewrite

**Comment by reduz** [comment]
This was effectively fixed by #7747
